cell_classification/augmentation_pipeline.py

```diff
@@ -137,3 +138,351 @@ def py_aug(batch, tf_aug):
+        """Returns 2D Gaussian kernel for convolutions.
+        Args:
+            sigma (float):
+                Standard deviation of the gaussian filter
+            n_channels (int):
+                The number of channels of the image
+        Returns:
+            tf.Tensor:
+                The gaussian kernel
+        """
+        x = tf.range(-self.size, self.size + 1, dtype=tf.float32)
+        g = tf.math.exp(-(x ** 2) / (2.0 * sigma ** 2))
+        g_kernel = tf.tensordot(g, g, axes=0)
+        g_kernel = g_kernel / tf.reduce_sum(g_kernel)
+        g_kernel = tf.expand_dims(g_kernel, axis=-1)
+        g_kernel = tf.expand_dims(g_kernel, axis=-1)
+        g_kernel = tf.tile(g_kernel, [1, 1, n_channels, 1])
+        return g_kernel
+
+    def __call__(self, image, labels):
+        """ Applies gaussian blurring
+        Args:
+            image (tf.Tensor):
+                The image to blur
+            labels (tf.Tensor):
+                The labels
+        Returns:
+            tf.Tensor:
+                The blurred image
+            tf.Tensor:
+                The labels
+        """
+        if tf.rank(image) == 3:
+            image = tf.expand_dims(image, axis=0)
+        if tf.random.uniform(()) < self.prob:
+            sigma = tf.random.uniform((), self.min_std, self.max_std)
+            kernel = self.gaussian_kernel(sigma, n_channels=tf.shape(image)[-1])
+            image = tf.nn.depthwise_conv2d(
+                image, kernel, strides=[1, 1, 1, 1], padding='SAME'
+            )
+        image = tf.squeeze(image)
+        return image, labels
+
+
+class Zoom(tf.Module):
+    """ Zoom augmentation"""
+    def __init__(self, prob=0.5, min_zoom=0.8, max_zoom=1.2, fill_mode='constant'):
+        """
+        Args:
+            prob (float):
+                The probability of applying the augmentation
+            min_zoom (float):
+                The minimum zoom factor
+            max_zoom (float):
+                The maximum zoom factor
+        """
+        super(Zoom, self).__init__()
+        self.prob = prob
+        self.min_zoom = min_zoom
+        self.max_zoom = max_zoom
+        self.fill_mode = fill_mode
+
+    def __call__(self, image, labels):
+        """ Applies zoom
+        Args:
+            image (tf.Tensor):
+                The image to zoom
+            labels (tf.Tensor):
+                The labels
+        Returns:
+            tf.Tensor:
+                The zoomed image
+            tf.Tensor:
+                The labels
+        """
+        if tf.random.uniform(()) < self.prob:
+            zoom_factor = tf.random.uniform([1], self.min_zoom, self.max_zoom)
+            zoom_factor = 1/zoom_factor
+            zooms = tf.expand_dims(
+                tf.cast(
+                    tf.concat([zoom_factor, zoom_factor], axis=0), dtype=tf.float32
+                ), 0
+            )
+            squeeze = False
+            if tf.rank(image) == 3:
```

Comment:
The `tf.map` function in augmenter squeezes the batch dimension. Since we'll probably use the augmentations only via `Augmenter` and thus `tf.map` I could get rid of the control flow and just assume HWC everywhere inside the augmentations.